common/config/loader.go

```diff
@@ -90,13 +98,37 @@ func Load(env string, configDir string, zone string, config interface{}) error {
 	// TODO: remove log dependency.
 	stdlog.Printf("Loading config files=%v\n", files)
 
+	templateFuncs := sprig.FuncMap()
+
 	for _, f := range files {
 		// This is tagged nosec because the file names being read are for config files that are not user supplied
 		// #nosec
 		data, err := os.ReadFile(f)
 		if err != nil {
 			return err
 		}
+
+		// If the config file contains "enable-template" in a comment within the first 1KB, then
+		// we will treat the file as a template and render it.
+		templating, err := checkTemplatingEnabled(data)
+		if err != nil {
+			return err
+		}
+
+		if templating {
+			tpl, err := template.New("config").Funcs(templateFuncs).Parse(string(data))
```

Comment:
nit: instead of `"config"` should we use `filepath.Base(f)`? it might produce nicer error messages?

Reply:
The filename also appears elsewhere in the error, but I've done this anyway.